### PR TITLE
✨ enable extension modal

### LINKED ISSUE #2516: Make non-blocking connection to the extensions
Currently the app is blocked when until you don't accept/reject extensions. 

Screenshot 2022-03-05 at 10 29 09

The good approach would be to only request extension when user clicks on it. 

The edge case is that we should somehow cache the last logged wallet (e.g which extenstion and address you used)

So using any storage (vuex, localStorage would be a good option) // I prefer Auth vuex here

Ref #2480

_Originally posted by @vikiival in https://github.com/kodadot/nft-gallery/issues/2482#issuecomment-1060368537_

### PR DESCRIPTION
**Pull request by roiLeo**
continue #2518, tested it, wasn't working but now it does

### PR type

- [x] Bugfix

### What's new?

- [x] PR closes #2516 
- [x] auto enable extension was coming from `plugins/globalVariable.ts` file so I removed it

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=HguGafrfk8nRP9a4tgJf7hLvSSFie8GtkDjyWFHQ2kDUAMc)

### Screenshot

**insert video here**

### CODE REVIEW

**Comment by vikiival** [comment]
> might have all extension on hand?

Locked and loaded! 🔫

File: components/common/WalletModal.vue [hunk 1]
```diff
@@ -92,9 +92,7 @@ import { Component, Prop, Vue } from 'nuxt-property-decorator'
 import { SupportedWallets, Wallet, WalletAccount } from '@/utils/config/wallets'
 
 @Component({
-  components: {
-    AccountSelect: () => import('@/components/shared/AccountSelect.vue'),
-  },
+  components: {},
```
Comment (vikiival) [inline components/common/WalletModal.vue:95]:
```suggestion

```

File: components/common/WalletModal.vue [hunk 2]
```diff
@@ -108,6 +106,7 @@ export default class WalletModal extends Vue {
   set account(account: string) {
     this.$emit('close')
     this.$store.dispatch('setAuth', { address: account })
+    localStorage.setItem('selected_account', account)
```
Comment (vikiival) [inline components/common/WalletModal.vue:109]: Something better than snek case? `selected_account`

like: `kodauth`, `auth`, `selectedAccount`

File: utils/extension.ts [hunk 3]
```diff
@@ -12,6 +13,20 @@ export const getAddress = async (address: string) => {
   }
 }
 
+export const getSelectedAccount = (accounts: WalletAccount[]) => {
```
Comment (vikiival) [inline utils/extension.ts:16]: Where is this used?

File: components/common/WalletModal.vue (hunk 2 above)
Comment (vikiival) [reply components/common/WalletModal.vue:109]: Also I would move it to the constants or something :) 
One typo aaand It's gone

Comment (roiLeo) [reply components/common/WalletModal.vue:109]: I've already had problems with camelCase key in localStorage, let me know wich key you want to use

**Comment by vikiival** [comment]
So the UX is far smoother
but It has a bug with the extensions:

First I approved polkadot-js (pJS)
Screenshot 2022-03-17 at 12 10 12

then went back and approved subwallet where I got pJS addresses
 
Screenshot 2022-03-17 at 12 10 46

then I approved talisman and still got pJS addresses


after when I went back again: the addresses were correct

File: utils/extension.ts (hunk 3 above)
Comment (roiLeo) [reply utils/extension.ts:16]: I wanted to use this in WalletModal.vue when we are setting wallet `l:135`, should I remove it?

File: components/common/WalletModal.vue (hunk 1 above)
Comment (roiLeo) [reply components/common/WalletModal.vue:95]: nope this will break stuff

**Comment by yangwao** [comment]
will this resolve https://github.com/kodadot/nft-gallery/issues/2480 ?

**Comment by roiLeo** [comment]
> will this resolve
> 
>     * [Website unresponsive if Talisman installed #2480](https://github.com/kodadot/nft-gallery/issues/2480) ?

It looks like #2480 and #2516 are somewhat the same

**Comment by yangwao** [comment]
noticed bug
when I go through dialogue.. ask for an account, polkadotjs extension want to authorize and I click authorize... I don't see accounts... until I do refresh. 

Can we refresh it somehow for users?

**Comment by roiLeo** [comment]
> noticed bug when I go through dialogue.. ask for an account, polkadotjs extension want to authorize and I click authorize... I don't see accounts... until I do refresh.
> 
> Can we refresh it somehow for users?

I managed to reproduce it once but I can't to make the select options reactive to whether or not the user as allowed the website in extension app for now

**Comment by vikiival** [comment]
> I can't to make the select options reactive to whether or not the user as allowed the website in extension app for now

So what if we like to give it a one more try after a couple of seconds?

**Comment by yangwao** [comment]
everyone has their tiny little secret I guess 


https://user-images.githubusercontent.com/5887929/158826459-96fb12e5-f35e-44ed-9a13-e860013764ce.mov


image

**Comment by yangwao** [comment]
me nudging @vikiival sitting next to me in CBD shop

**Comment by vikiival** [comment]
Works as a charm!
jeoqzqetwyu51

Except the SubWallet


Screenshot 2022-03-19 at 11 54 40

**Comment by yangwao** [comment]
Apparently Subwallet doesn't work here, let's crack on this why in another issue to push this forward :)

**Comment by yangwao** [comment]
pay 200 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $200 @ 145.08 USD/KSM ~ 1.379 $KSM
🧗 [HguGafrfk8nRP9a4tgJf7hLvSSFie8GtkDjyWFHQ2kDUAMc](https://kusama.subscan.io/account/HguGafrfk8nRP9a4tgJf7hLvSSFie8GtkDjyWFHQ2kDUAMc)
🔗 [0x2444c332b26f1718e22a6a15eace1e7a728331108154142ca9fea6872c955ecd](https://kusama.subscan.io/extrinsic/0x2444c332b26f1718e22a6a15eace1e7a728331108154142ca9fea6872c955ecd)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**